cli/start.go

```diff
@@ -121,6 +124,42 @@ func initInsecure() error {
 	return nil
 }
 
+func initMemProfile(dir string) {
+	memProfileInterval := envutil.EnvOrDefaultDuration("memprof_interval", -1)
+	if memProfileInterval == -1 {
+		return
+	}
+	if min := time.Second; memProfileInterval < min {
+		log.Infof("fixing excessively small memory profiling interval: %s -> %s",
+			memProfileInterval, min)
+		memProfileInterval = min
+	}
+
+	log.Infof("writing memory profiles to %s every %s", dir, memProfileInterval)
+
+	go func() {
+		t := time.NewTicker(memProfileInterval)
+		for {
+			<-t.C
+
+			func() {
+				const format = "2006-01-02T15_04_05.999"
+				path := filepath.Join(dir, "memprof."+timeutil.Now().Format(format))
+				f, err := os.Create(path)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "testing: %s\n", err)
+					os.Exit(2)
+				}
+				defer f.Close()
+				if err = pprof.WriteHeapProfile(f); err != nil {
+					log.Warningf("can't write %s: %s\n", path, err)
+				}
+				log.Infof("%s", path)
```

Comment:
Yes, I meant to keep this as it logs whenever a memory profile is being written. Not sure of the utility though as you can simply see the files being created. Thoughts?